src/writer.cpp

```diff
@@ -166,42 +167,52 @@ namespace sdptransform
 
 		linestream << type << "=";
 
+		string_view runningFormat(format);
+		size_t nextPos = 0U;
 		for(
-			auto it = std::sregex_iterator(format.begin(), format.end(), FormatRegex);
-			it != std::sregex_iterator();
-		  ++it
-		 )
+			auto pos = runningFormat.find('%'); pos != std::string::npos; pos = runningFormat.find('%',nextPos))
 		{
-			const std::smatch& match = *it;
-			const std::string& str = match.str();
+			const char nextChar = (pos+1<runningFormat.size()) ? runningFormat[pos+1] : '0';
+			const string_view prefix = runningFormat.substr(0,pos);
+
+			if(nextChar != '%' && nextChar != 'v' && nextChar != 'd' && nextChar != 's') {
+				nextPos = pos +2;
+				continue;
+			}
 
 			if (i >= len)
 			{
-				linestream << str;
+				linestream << prefix << '%' << nextChar;
 			}
 			else
 			{
 				auto& arg = args[i];
 				i++;
 
-				linestream << match.prefix();
+				linestream << prefix;
 
-				if (str == "%%")
+				if (nextChar == '%')
 				{
 					linestream << "%";
 				}
-				else if (str == "%s" || str == "%d")
+				else if (nextChar == 's' || nextChar == 'd')
 				{
 					if (arg.is_string())
 						linestream << arg.get<std::string>();
 					else
 						linestream << arg;
 				}
-				else if (str == "%v")
+				else if (nextChar == 'v')
 				{
 					// Do nothing.
 				}
 			}
+
+			runningFormat=runningFormat.substr(pos+2);
+			nextPos=0;
+		}
+		if(!runningFormat.empty()) {
```

Comment:
Those changes do not respect code syntax.

Reply:
what is the code syntax you want to review ?

Reply:
Same one as in the existing code, including blank lines before and after condition block, spaces before if, opening brackets in new line, etc.

Reply:
what tool do you use for formatting ? clang-format ? vscode ?

Reply:
No one in this project unfortunately. But we do in mediasoup and libmediasoupclient with clang-format.

Reply:
ok I'll launch clang-format on it :)